### ISSUE TITLE
[1.4.2.6] nested node_modules from a meteor package not being installed

### ISSUE THREAD

**Issue by sebakerckhof** [meteor/meteor#8349]
When upgrading from meteor 1.4.2.6 my application started to throw errors about modules not being found.

Upon inspection I found the module was a module that was a dependency of another npm package that was a dependency in `package.js` file of a local meteor package.
Downgrading to 1.4.2.3 fixes the problem.

So it's like this:
Meteor package -> Npm package 1 -> Npm Package 2
Npm package 2 is not being installed with meteor 1.4.2.6, but it is in meteor 1.4.2.3

The `npm-shrinkwrap.json` file inside the `package/.npm` folder contains only 1 entry: the direct dependency listed in the `package.js` file when using meteor 1.4.2.6. The one when using meteor 1.4.2.3 contains a whole list of entries

**Comment by sebakerckhof** [comment]
Update: strangely enough, I only seem to have this problem when using my private npm server.

So situation is like this:
 - Private NPM server + Meteor 1.4.2.3 -> OK
 - Public NPM server + Meteor 1.4.2.6 -> OK
 - Public NPM server + Meteor 1.4.2.6 -> NOT OK

**Comment by benjamn** [comment]
Are these local Meteor packages? What happens if you delete the `.npm` directory entirely (should force it to be recreated)?

**Comment by sebakerckhof** [comment]
Yep, local meteor package. The behavior described above was by each time doing a `meteor reset` and deleting the `package/.npm` folder in between. 
However, I've done some more tests now and now I was able to get it to work on 1.4.2.6. So it seems to be not reliably reproducible and therefore my findings above might not be correct.

As a side note, I just notice that the npm-shirnkwrap looks something like:
```
...
"azure-storage": {
      "version": "0.10.0",
      "resolved": "https://bin.barco.com/artifactory/api/npm/npm-virtual/azure-storage/-/azure-storage-0.10.0.tgz",
      "from": "azure-storage@0.10.0"
    },
    "bcrypt-pbkdf": {
      "version": "1.0.1",
      "resolved": "https://registry.npmjs.org/bcrypt-pbkdf/-/bcrypt-pbkdf-1.0.1.tgz",
      "from": "bcrypt-pbkdf@>=1.0.0 <2.0.0"
    },
    "bl": {
      "version": "1.0.3",
      "resolved": "https://registry.npmjs.org/bl/-/bl-1.0.3.tgz",
      "from": "bl@>=1.0.0 <1.1.0"
    },
...
```

`azure-storage` is the package defined in `package.js` and `bin.barco.com` is my private NPM server.

The other packages are dependencies of the `azure-storage` package. So this means the private NPM server settings are only used for the top-level dependency. But that's a different problem.

**Comment by StorytellerCZ** [comment]
I have similar problem as well. When trying to deploy to Galaxy about a node module not found (`detect-indent`). This did not appear when I tried to test run Meteor build with `--production`. After deleting the `node_modules` folder and installing it anew the problem went away for me.

**Comment by kubami** [comment]
I had a same problem when trying to deploy with `pm2-meteor`.
I have managed to fix it with 
- removing `node_modules`
- reinstalling all of them with `meteor npm install`

**Comment by sebakerckhof** [comment]
@StorytellerCZ and @kubami are you on Meteor 1.4.2.6+? Because there seems to have been a fix to a similar issue that appeared in 1.4.2.5.

**Comment by abernix** [comment]
@sebakerckhof @StorytellerCZ @kubami Meteor 1.4.2.7 ( #8361 ) is currently published, but not (yet) recommended and rolls back the `npm` upgrade which caused this issue.  Any chance either of you could confirm this is fixed in 1.4.2.7?

In addition!  Meteor 1.4.3.1 is also published (and likewise, not recommended, yet!), and contains a more complete fix (#8363)  to allow us to upgrade to NPM 4 and working with [the new shrinkwrap mechanism](https://github.com/npm/npm/blob/latest/CHANGELOG.md#no-more-partial-shrinkwraps-breaking) it enforces.  If you can provide any confirmation that it's working as you expect, that would also be appreciated!

**Comment by kubami** [comment]
@abernix My problems arised when I updated meteor through `meteor update`. It was a 1.4.2.3 -> 1.4.2.6 update. I had the same symptoms as @sebakerckhof.
I have since fixed it by removing `node_modules` and reinstalling all npm deps `npm install`.

It would seem that it's not really a bug but an issue with updating process.

I have tried 1.4.3.1 now and it works perfectly :).
I will try to try 1.4.2.7 later.
